engines/groovie/video/roq.cpp

```diff
@@ -602,9 +617,41 @@ bool ROQPlayer::processBlock() {
 	}
 
 	if (endpos != _file->pos() && !_file->eos()) {
-		warning("Groovie::ROQ: BLOCK %04x Should have ended at %lld, and has ended at %lld", blockHeader.type, (long long)endpos, (long long)_file->pos());
+		warning("Groovie::ROQ: BLOCK 0x%04x Should have ended at %lld, and has ended at %lld", blockHeader.type, (long long)endpos, (long long)_file->pos());
 		warning("Ensure you've copied the files correctly according to the wiki.");
-		_file->seek(MIN(_file->pos(), endpos));
+		int64 currFilePos = _file->pos();
+		if (currFilePos < endpos) {
+			// In this case we stay at the current position,
+			// which is *before* the intended endpos (as calculated from its blockHeader.size) of the faulty block
+
+			// We "peek" ahead to see what type of block is next
+			bool foundValidBlockHeaderType = false;
+			uint16 blockHeader_type = _file->readUint16LE();
```

Comment:
This should be camelCase, like the rest of the code
```suggestion
			uint16 blockHeaderType = _file->readUint16LE();
```

Reply:
done.